core/src/main/java/org/infinispan/transaction/tm/DummyTransaction.java

```diff
@@ -45,7 +47,7 @@
    private final Xid xid;
    private volatile int status = Status.STATUS_UNKNOWN;
    private final List<Synchronization> syncs;
-   private final List<XAResource> resources;
+   private final Map<XAResource, Integer> resources;
```

Comment:
BTW the 1st commit uses 2 collections, the second commit is optional to use only 1 collection.